### DISCUSSION TITLE
투자미션 새로고침 관련해서

### DISCUSSION THREAD

**Discussion by mille-feu** [Tiny-Metro/Poly-Metro#197] [Q&A]
이하 내용이 기능의 적절한 이해가 맞는지 확인이 필요합니다.
- 게임 시작 시 이용 가능한 투자미션은 1개 (랜덤 생성)
- 1주일마다 최대 3개까지 이용 가능한 투자미션이 늘어남 (랜덤 생성)
- 3개가 된 이후 (4주차부터) 새 투자미션 생성 시, 기존의 처음 (4주차의 경우 1주차의 투자미션) 투자미션이 삭제됨
- 위와 같이 매 주 새로 한 개의 투자미션이 새로 생성, 가장 오래된 투자미션 삭제
- 리프레시 버튼을 누를 경우, 모든 투자미션이 새로 생성되고, 이 때 중복된 미션은 나타나지 않음

<br>

이하 내용에 대한 답이 필요합니다.
- 이용 가능한 모든 투자미션이 진행중일 때, 리프레시 버튼 혹은 다음 주차로 넘어갈 경우 어떻게 되는가?
- 다음 주차로 넘어갈 때, 가장 오래된 투자미션이 진행중일 경우 그 다음 오래된 투자미션을 삭제하는게 맞는가?

**Comment by seong0929** [comment]
- 이용 가능한 모든 투자미션이 진행중일 때, 리프레시 버튼 혹은 다음 주차로 넘어갈 경우 어떻게 되는가?
모든 투자 미션이 진행중이면, 미션 중 한 미션이라도 끝나지 않으면 리프레시 버튼 혹은 다음 주차로 넘어가도 변경이 없습니다.

- 다음 주차로 넘어갈 때, 가장 오래된 투자미션이 진행중일 경우 그 다음 오래된 투자미션을 삭제하는게 맞는가?
다음 주차로 넘어갈 때, 가장 오래된 투자미션이 진행중이면 그 다음 오래된 투자미션이 삭제되며, 전부 진행 중이면 삭제되지 않고, 리프레시도 되지 않습니다.

**Reply by mille-feu** [reply]
답변 감사합니다.
추가로 아래의 경우는 어떤 식으로 처리되나요?
- 진행중인 투자미션이 완료되었을 경우 (실패, 성공 무관)
  - 완료된 미션이 바로 삭제되고 다른 미션이 생성됨
  - 완료처리로 표시 후, 리프레시 혹은 다음 주차에 삭제 및 교체됨

**Reply by seong0929** [reply]
후자가 맞습니다. 정확히는 진행중인 투자미션이 완료되었을 경우 (실패, 성공 무관) 완료처리로 표시 후, 리프레시 혹은 다음 주차에 교체됨이 맞겠네요
